build: Build fails on VS2013 and earlier

<!--
If you have a question rather than reporting a bug please go to http://answers.opencv.org where you get much faster responses.
If you need further assistance please read [How To Contribute](https://github.com/opencv/opencv/wiki/How_to_contribute).

This is a template helping you to create an issue which can be processed as quickly as possible. This is the bug reporting section for the OpenCV library.
-->

##### System information (version)
<!-- Example
- OpenCV => 3.1
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2015
-->

- OpenCV => master ( a391871ccea84d7fa4a672ae1be082c505d7ff1d )
- Operating System / Platform => Windows 10 64bit
- Compiler => Visual Studio 2012 / 2013

##### Detailed description

Building OpenCV using Visual Studio 2013 and earlier fails in error

```

  C:\work\OpenCV-MSBuild\modules\dnn\test\npy_blob.cpp(75): error C2065: 'uint16_t' : undeclared identifier [C:\work\OpenCV-MSBuild\build\modules\dnn\opencv_test_dnn.vcxproj]
  C:\work\OpenCV-MSBuild\modules\dnn\test\npy_blob.cpp(75): error C2146: syntax error : missing ';' before identifier 'headerSize' [C:\work\OpenCV-MSBuild\build\modules\dnn\opencv_test_dnn.vcxproj]
  C:\work\OpenCV-MSBuild\modules\dnn\test\npy_blob.cpp(75): error C2065: 'headerSize' : undeclared identifier [C:\work\OpenCV-MSBuild\build\modules\dnn\opencv_test_dnn.vcxproj]
  C:\work\OpenCV-MSBuild\modules\dnn\test\npy_blob.cpp(76): error C2065: 'headerSize' : undeclared identifier [C:\work\OpenCV-MSBuild\build\modules\dnn\opencv_test_dnn.vcxproj]
  C:\work\OpenCV-MSBuild\modules\dnn\test\npy_blob.cpp(76): error C2070: ''unknown-type'': illegal sizeof operand [C:\work\OpenCV-MSBuild\build\modules\dnn\opencv_test_dnn.vcxproj]
  C:\work\OpenCV-MSBuild\modules\dnn\test\npy_blob.cpp(78): error C2065: 'headerSize' : undeclared identifier [C:\work\OpenCV-MSBuild\build\modules\dnn\opencv_test_dnn.vcxproj]
```

Basically the error is missing definition of uint16_t on Visual Studio 2013 and earlier. The situation is very simillar to #8868.
I'll write a PR later


##### Steps to reproduce

Build OpenCV using Visual Studio 2012 and 2013.  Make sure the dnn is included in the build target

--- a/modules/dnn/test/npy_blob.cpp
+++ b/modules/dnn/test/npy_blob.cpp
@@ -72,7 +72,7 @@ Mat blobFromNPY(const std::string& path)
     ifs.ignore(1);  // Skip major version byte.
     ifs.ignore(1);  // Skip minor version byte.
 
-    uint16_t headerSize;
+    unsigned short headerSize;
     ifs.read((char*)&headerSize, sizeof(headerSize));
 
     std::string header(headerSize, '*');

